### PR TITLE
#2574 Added show only encrypted emails filter buttoN

### LINKED ISSUE #2574: Add "only encrypted emails" filter
There was a customer request to have ability to show only encrypted emails.

Let's add `show only encrypted messages` filter to the search screen, tapping on it will show only encrypted messages on search results screen.

Earlier similar functionality was implemented on Android, but was disabled as there were some issues - https://github.com/FlowCrypt/flowcrypt-android/issues/715

I think we can use [users.message.list](https://developers.google.com/gmail/api/reference/rest/v1/users.messages/list?apix_params=%7B%22userId%22%3A%22me%22%2C%22q%22%3A%22BEGIN+PGP+MESSAGE%22%7D) endpoint with `q` parameter for filtering encrypted messages.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR added show only encrypted emails filter button

close #2574 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Controllers/Inbox/InboxViewController.swift [hunk 1]
```diff
@@ -243,11 +258,16 @@ extension InboxViewController {
...
 extension InboxViewController {
     private func getSearchQuery() -> String? {
-        guard searchedExpression.isNotEmpty else { return nil }
+        let showOnlyPgp = UserDefaults.standard.bool(forKey: "SHOW_PGP_ONLY_FLAG")
+        let pgpPattern = """
+            ("-----BEGIN PGP MESSAGE-----" AND "-----END PGP MESSAGE-----") OR \
+            ("-----BEGIN PGP SIGNED MESSAGE-----") OR \
+            filename:({asc pgp gpg key})
+        """
+        let baseQuery = showOnlyPgp ? "\(pgpPattern) AND \(searchedExpression)" : searchedExpression
```
Comment (sosnovsky) [inline FlowCrypt/Controllers/Inbox/InboxViewController.swift:267]: currently `onlyPgp` mode works correctly only for the initial message loading.
when you scroll to the bottom of messages list and next page is loaded - it loads plain messages too:
Screenshot 2024-11-08 at 11 10 22

File: FlowCryptUI/Nodes/PgpOnlySwitchNode.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+public final class PgpOnlySwitchNode: CellNode {
+
+    let SHOW_PGP_ONLY_KEY = "SHOW_PGP_ONLY_FLAG"
+
+    private lazy var imageNode: ASImageNode = {
+        let node = ASImageNode()
+        node.image = UIImage(systemName: "lock.shield")?.tinted(.main)
+        node.style.preferredSize = CGSize(width: 30, height: 30)
+        return node
+    }()
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCryptUI/Nodes/PgpOnlySwitchNode.swift]: Sounds good!

File: FlowCrypt/Controllers/Inbox/InboxViewController.swift (hunk 1 above)
Comment (ioanmo226) [reply FlowCrypt/Controllers/Inbox/InboxViewController.swift:267]: Yep. Nice catch :+1:

**Review by sosnovsky — COMMENTED** [review]
Works great now, thanks!

Just a minor UI issue - switch element seems to cropped by a couple pixles on the right side:

Screenshot 2024-11-08 at 14 49 50

**Comment by ioanmo226** [comment]
Fixed. Please check

**Review by sosnovsky — APPROVED** [review]
All good now, thanks!

**Pull request merged**